core/domain/suggestion_registry.py

```diff
@@ -315,6 +315,9 @@ def populate_old_value_of_change(self) -> None:
             'Subclasses of BaseSuggestion should implement '
             'populate_old_value_of_change.')
 
+    # Here we use type Any because sub-classes of BaseSuggestion can have
```

Comment:
This does not seem correct, at the very least it should be a Union type.

Reply:
I think I have explained before -- that we cannot define Union in the superclass.

https://mypy-play.net/?mypy=latest&python=3.10&flags=strict%2Cdisallow-any-expr&gist=df9686c78b9ef983d6c1892b5b6e3e8c 

Please run the example that I have put above -- by clicking on the `run` button to see the error.

Reply:
That makes sense -- thanks, that's a very helpful reply. I read up on the various principles and understand the issue.

My question then becomes: how do other codebases handle it? Surely we aren't the first to tackle subclasses in mypy. Do they use Any?

Reply:
The most closet answer that I have found is https://stackoverflow.com/a/54359797

Do they use Any? -- yes some of them.

Reply:
The most closet answer that I have found is https://stackoverflow.com/a/54359797

Do they use Any? -- yes some of them.

And i don't see any problem using Any here -- because we are allowing for sub-classes and not doing this for any implementation purpose like we have done in customization arg one!

Reply:
TODO is added with detailed explanation.